core/src/test/java/io/questdb/test/cutlass/http/line/LineHttpSenderTest.java

```diff
@@ -0,0 +1,386 @@
+                serverMain.assertSql("SELECT sum(water_level) FROM h2o_feet", "sum\n" + (count * (count - 1) / 2) + "\n");
+            }
+        });
+    }
+
+    @Test
+    public void testLineHttpDisabled() throws Exception {
+        Rnd rnd = TestUtils.generateRandom(LOG);
+        TestUtils.assertMemoryLeak(() -> {
+            int fragmentation = 1 + rnd.nextInt(5);
+            LOG.info().$("=== fragmentation=").$(fragmentation).$();
+            try (final TestServerMain serverMain = startWithEnvVariables(
+                    LINE_HTTP_ENABLED.getEnvVarName(), "false"
+            )) {
+                int httpPort = getHttpPort(serverMain);
+
+                int totalCount = 1_000;
+                try (LineHttpSender sender = new LineHttpSender("localhost", httpPort, DefaultHttpClientConfiguration.INSTANCE, null, 100_000, null, null, null, 0)) {
+                    for (int i = 0; i < totalCount; i++) {
+                        sender.table("table")
+                                .longColumn("lcol1", i)
+                                .atNow();
+                    }
+                    try {
+                        sender.flush();
+                        Assert.fail("Expected exception");
+                    } catch (LineSenderException e) {
+                        TestUtils.assertContains(e.getMessage(), "http-status=404");
+                        TestUtils.assertContains(e.getMessage(), "Could not flush buffer: HTTP endpoint does not support ILP.");
+                    }
+                }
+            }
+        });
+    }
+
+    @Test
+    public void testRestrictedCreateColumnsError() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            try (final TestServerMain serverMain = startWithEnvVariables(
+                    PropertyKey.HTTP_RECEIVE_BUFFER_SIZE.getEnvVarName(), "2048",
+                    PropertyKey.LINE_AUTO_CREATE_NEW_COLUMNS.getEnvVarName(), "false"
+            )) {
+                serverMain.start();
+                serverMain.compile("create table ex_tbl(b byte, s short, f float, d double, str string, sym symbol, tss timestamp, " +
+                        "i int, l long, ip ipv4, g geohash(4c), ts timestamp) timestamp(ts) partition by DAY WAL");
+
+                int port = IlpHttpUtils.getHttpPort(serverMain);
+                try (Sender sender = Sender.builder()
+                        .url("http://localhost:" + port)
+                        .build()
+                ) {
+                    sender.table("ex_tbl")
+                            .symbol("a3", "3")
+                            .at(1222233456, ChronoUnit.NANOS);
+                    flushAndAssertError(
+                            sender,
+                            "Could not flush buffer",
+                            "http-status=400",
+                            "error in line 1: table: ex_tbl, column: a3 does not exist, creating new columns is disabled"
```

Comment:
good idea, let me change it

Reply:
it seems like this is not ILP@HTTP specific, so I prefer not to change it in this PR:
<img width="900" alt="image" src="https://github.com/questdb/questdb/assets/158619/63b44bf3-6fc8-41cd-885b-2fbd120530b7">

ok with you?

Reply:
In case of ILP/HTTP, that error text is returned to the client. In other cases, it's just a log message. In any case, let's keep it as is if you don't want to touch that code.